src/main/java/com/jfrog/ide/common/scan/ScanManagerBase.java

```diff
@@ -113,7 +109,7 @@ protected Artifact getArtifactSummary(String componentId) {
      */
     public void scanAndCacheArtifacts(ProgressIndicator indicator, boolean quickScan) throws IOException, InterruptedException {
         log.debug("Start scan for '" + projectName + "'.");
```

Comment:
Should we add a null check and a more informative exception message?
(because now an initialize call to setScanLogic is mandatory)

Reply:
I'm not sure, because this flow should never happen. 
From the user's perspective, it doesn't matter if it is NPE or a more informative message. The stacktrance should be sufficient for us to fix it.
From the developer's perspective, adding an informative message to the stacktrace will not help to resolve the issue.